src/kvstore/NebulaStore.cpp

```diff
@@ -1051,6 +1056,58 @@ void NebulaStore::cleanWAL() {
     }
 }
 
+ErrorOr<ResultCode, std::vector<std::string>> NebulaStore::backupTable(
+    GraphSpaceID spaceId,
+    const std::string& name,
+    const std::string& tablePrefix,
+    std::function<bool(const folly::StringPiece& key)> filter) {
+    auto spaceRet = space(spaceId);
+    if (!ok(spaceRet)) {
+        return error(spaceRet);
+    }
+
+    auto space = nebula::value(spaceRet);
+    std::vector<std::string> backupPath;
+    for (auto& engine : space->engines_) {
+        auto path = engine->backupTable(name, tablePrefix, filter);
+        if (!ok(path)) {
+            auto result = error(path);
+            if (result != kvstore::ResultCode::ERR_BACKUP_EMPTY_TABLE) {
+                return result;
+            }
+            LOG(WARNING) << "Since the table(" << tablePrefix
+                         << ") is empty, the backup of the current table is skipped.";
+            continue;
+        }
+        backupPath.emplace_back(value(path));
+    }
+
+    if (backupPath.empty()) {
+        return kvstore::ResultCode::ERR_BACKUP_EMPTY_TABLE;
+    }
+
+    return backupPath;
+}
+
+ResultCode NebulaStore::restoreFromFiles(GraphSpaceID spaceId,
+                                         const std::vector<std::string>& files) {
+    auto spaceRet = space(spaceId);
+    if (!ok(spaceRet)) {
+        LOG(ERROR) << "Get Space " << spaceId << " Failed";
+        return error(spaceRet);
+    }
+    auto space = nebula::value(spaceRet);
+
+    for (auto& engine : space->engines_) {
```

Comment:
`restoreFromFiles` is both called by storage and meta? If storage call this method, it could be wrong if multi data_path is specified.

Reply:
"restoreFromFiles" will only be called by Meta.